dvc/repo/fetch.py

```diff
@@ -81,20 +81,43 @@ def fetch(
 def _fetch_external(self, repo_url, repo_rev, files, jobs):
     from dvc.external_repo import external_repo
     from dvc.scm.base import CloneError
+    from dvc.path_info import PathInfo
+
+    failed = 0
+
+    results = []
+
+    def cb(result):
+        results.append(result)
 
-    failed, downloaded = 0, 0
     cache = self.cache.local
     try:
         with external_repo(
             repo_url, repo_rev, cache_dir=cache.cache_dir
         ) as repo:
-            d, f, _ = repo.fetch_external(files, jobs=jobs)
-            downloaded += d
-            failed += f
+            root = PathInfo(repo.root_dir)
+            for path in files:
+                path_info = root / path
+                hash_info = repo.repo_tree.get_hash(
+                    path_info,
+                    download_callback=cb,
+                    follow_subrepos=False,
+                    jobs=jobs,
+                )
+                if not hash_info:
+                    raise FileNotFoundError
+                self.cache.local.save(
+                    path_info,
+                    repo.repo_tree,
+                    hash_info,
+                    link=False,
+                    save_link=False,
+                    download_callback=cb,
+                )
```

Comment:
See https://github.com/iterative/dvc/pull/5234/files#r553347502 . This will likely be something like `cache.transfer()` in the future (or maybe `cache.save()` will start returning the hash).